guided-charts/aws-hyperpod/templates/network-policies.yaml

```diff
@@ -0,0 +1,50 @@
+{{- if .Values.networkPolicy.enabled }}
+---
+# NetworkPolicy to restrict access to auth middleware
+# Only Traefik can communicate with auth middleware
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: authmiddleware-network-policy
+  namespace: {{ .Values.namespace }}
+spec:
+  podSelector:
+    matchLabels:
+      app: authmiddleware
+      component: auth
+  policyTypes:
+  - Ingress
+  ingress:
+  - from:
+    - podSelector:
```

Comment:
consider requiring that the traffic comes from the same namespace as well.

Example in #132